src/component-helpers.js

```diff
@@ -28,8 +32,13 @@ export function initialize(component) {
     throw new Error('Etch components must implement `update(props, children)`.')
   }
 
+  let virtualElement = component.render()
+  if (!validVirtualElement(virtualElement)) {
+    let namePart = component.constructor && component.constructor.name ? ' in ' + component.constructor.name : ''
+    throw new Error('invalid falsy value ' + virtualElement + ' returned from render()' + namePart)
```

Comment:
As I mentioned in the comment on the PR, don't all these things still need to be passed to `etch.dom`? That function only returns instances of `VNode` or `VText` [from the underlying virtual-dom library](https://github.com/Matt-Esch/virtual-dom/tree/master/vnode).